src/main/app.js

```diff
@@ -71,8 +72,33 @@ app.on('open-file', (e, filePath) => {
 });
 
 // Open file using Buttercup (on Windows)
-if (isWindows() && typeof process.argv[1] === 'string') {
-  initialFile = process.argv[1];
+if (isWindows()) {
+  initialFile = getFilePathFromArgv(process.argv);
+}
+
+// Someone tried to run a second instance, we should focus our window.
+const isSecondInstance = app.makeSingleInstance(argv => {
+  const handleArgvForWindow = win => {
+    // Handle the argv of second instance for windows
+    const filePath = getFilePathFromArgv(argv);
+    if (isWindows() && filePath) {
+      loadFile(filePath, win);
+    }
+  };
+  if (windowManager.getCountOfType('main') > 0) {
+    const [mainWindow] = windowManager.getWindowsOfType('main');
+    if (mainWindow.isMinimized()) {
+      mainWindow.restore();
+    }
+    mainWindow.focus();
+    handleArgvForWindow(mainWindow);
+  } else {
+    windowManager.buildWindowOfType('main', win => handleArgvForWindow(win));
+  }
+});
+
+if (isSecondInstance) {
+  app.quit();
```

Comment:
Hmm, will this create weird behaviour? Would it be possible to focus the first one? I guess that's probably quite difficult..

Reply:
Thats what the first few lines do. This is straight out of electron’s docs

Reply:
![41c1b875-6b94-426a-b022-9d3f62e27d4d](https://user-images.githubusercontent.com/768052/36565459-340fc08a-1829-11e8-8051-c1730ffa0a35.jpeg)

Reply:
Ah I missed those! Good stuff.